src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/SystemApiController.java

```diff
@@ -235,14 +235,36 @@ public SystemApiController(InternalTlsCertificateService internalTlsCertificateS
 
     @Override
     @PreAuthorize("hasAuthority('UPLOAD_ANCHOR')")
-    public ResponseEntity<Anchor> previewAnchor(Resource anchorResource) {
+    public ResponseEntity<Anchor> previewAnchor(Boolean verifyInstance, Resource anchorResource) {
         byte[] anchorBytes = ResourceUtils.springResourceToBytesOrThrowBadRequest(anchorResource);
         AnchorFile anchorFile = null;
         try {
-            anchorFile = systemService.getAnchorFileFromBytes(anchorBytes);
+            anchorFile = systemService.getAnchorFileFromBytes(anchorBytes, verifyInstance);
         } catch (SystemService.InvalidAnchorInstanceException | SystemService.MalformedAnchorException e) {
             throw new BadRequestException(e);
         }
         return new ResponseEntity<>(anchorConverter.convert(anchorFile), HttpStatus.OK);
     }
+
+    /**
+     * For uploading an initial configuration anchor. The difference between this and {@link #uploadAnchor(Resource)}
+     * is that the anchor's instance does not get verified
+     * @param anchorResource
+     * @return
+     */
+    @Override
+    @PreAuthorize("hasAuthority('INIT_CONFIG')")
+    public ResponseEntity<Void> uploadInitialAnchor(Resource anchorResource) {
+        byte[] anchorBytes = ResourceUtils.springResourceToBytesOrThrowBadRequest(anchorResource);
+        try {
+            systemService.uploadInitialAnchor(anchorBytes);
+        } catch (SystemService.InvalidAnchorInstanceException | SystemService.MalformedAnchorException
+                | SystemService.AnchorAlreadyExistsException e) {
```

Comment:
Should AnchorAlreadyExistsException be a conflict?

Reply:
It probably should! Changed